runtime/src/threads/pthreads/threads-pthreads.c

```diff
@@ -332,6 +409,15 @@ void chpl_thread_exit(void) {
     chpl_mem_free(tlp, 0, 0);
   }
 
+#ifdef CHPL_TASKS_FIFO_REGISTERED_STACK
+  while (thread_stack_list_head != NULL) {
```

Comment:
Shouldn't this loop be interacting with the lock you introduced - thread_stack_list_lock ?